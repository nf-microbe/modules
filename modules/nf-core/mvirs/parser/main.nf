process MVIRS_PARSER {
    tag "${meta.id}"
    label 'process_low'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'oras://community.wave.seqera.io/library/hmmer_prodigal-gv_biopython_pandas:de876e9113694da7' :
        'community.wave.seqera.io/library/hmmer_prodigal-gv_biopython_pandas:aaaf7b9a9207df90' }"

    input:
    tuple val(meta) , path(contigs)
    tuple val(meta2), path(prophages)
    tuple val(meta3), path(hmmsearch)
    tuple val(meta4), path(contigs_faa)

    output:
    tuple val(meta), path("${prefix}.mvirs.summary.tsv")    , emit: summary
    tuple val(meta), path("${prefix}.mvirs.integrases.tsv") , emit: integrases
    path "versions.yml"                                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args   ?: ''
    prefix   = task.ext.prefix ?: "${meta.id}"
    """
    mvirs_parser.py \\
        --mvirs ${prophages} \\
        --fna ${contigs} \\
        --hmmsearch ${hmmsearch} \\
        --faa ${contigs_faa} \\
        --prefix ${prefix}.mvirs \\
        ${args}

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
        pandas: \$(python -c "import pandas; print(pandas.__version__)")
    END_VERSIONS
    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.mvirs.summary.tsv
    touch ${prefix}.mvirs.integrases.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        python: \$( python --version | sed 's/Python //' )
        biopython: \$(python -c "import Bio; print(Bio.__version__)")
        pandas: \$(python -c "import pandas; print(pandas.__version__)")
    END_VERSIONS
    """
}
