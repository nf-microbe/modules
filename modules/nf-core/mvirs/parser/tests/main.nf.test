nextflow_process {

    name "Test Process MVIRS_PARSER"
    script "../main.nf"
    process "MVIRS_PARSER"

    tag "modules"
    tag "modules_nfcore"
    tag "mvirs"
    tag "mvirs/index"
    tag "mvirs/oprs"
    tag "mvirs/parser"

    test("fasta.gz + integrases.hmm") {
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file("${projectDir}/assets/test-datasets/genomics/provirus/illumina/fasta/test1.contigs.fasta.gz", checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test' ], // meta map
                    file("${projectDir}/assets/test-datasets/genomics/provirus/illumina/fasta/test1.mvirs.provirus.fasta.gz", checkIfExists: true)
                ]
                input[2] = [
                    [ id: 'test' ],
                    file("${projectDir}/assets/test-datasets/genomics/provirus/illumina/hmmsearch/test.hmmsearch", checkIfExists: true)
                ]
                input[3] = [
                    [ id: 'test' ],
                    file("${projectDir}/assets/test-datasets/genomics/provirus/illumina/fasta/test1.faa.gz", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("fasta.gz - stub") {
        options "-stub"
        when {
            process {
                """
                input[0] = [
                    [ id:'test' ], // meta map
                    file("${projectDir}/assets/test-datasets/genomics/provirus/illumina/fasta/test1.contigs.fasta.gz", checkIfExists: true)
                ]
                input[1] = [
                    [ id:'test' ], // meta map
                    file("${projectDir}/assets/test-datasets/genomics/provirus/illumina/fasta/test1.mvirs.provirus.fasta.gz", checkIfExists: true)
                ]
                input[2] = [
                    [ id: 'test' ],
                    file("${projectDir}/assets/test-datasets/genomics/provirus/illumina/hmmsearch/test.hmmsearch", checkIfExists: true)
                ]
                input[3] = [
                    [ id: 'test' ],
                    file("${projectDir}/assets/test-datasets/genomics/provirus/illumina/fasta/test1.faa.gz", checkIfExists: true)
                ]
                """
            }
        }
        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
