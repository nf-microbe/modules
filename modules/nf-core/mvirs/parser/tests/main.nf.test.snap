{
    "fasta.gz + integrases.hmm": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mvirs.summary.tsv:md5,9918439ea21b10132f6df7434c74107a"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mvirs.integrases.tsv:md5,545639e690399fe07da3cae9306d68f4"
                    ]
                ],
                "2": [
                    "versions.yml:md5,a98833d0eaa28fe37eb1685b05b4c8a8"
                ],
                "integrases": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mvirs.integrases.tsv:md5,545639e690399fe07da3cae9306d68f4"
                    ]
                ],
                "summary": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mvirs.summary.tsv:md5,9918439ea21b10132f6df7434c74107a"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a98833d0eaa28fe37eb1685b05b4c8a8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-25T11:20:18.759679614"
    },
    "fasta.gz - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mvirs.summary.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mvirs.integrases.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,a98833d0eaa28fe37eb1685b05b4c8a8"
                ],
                "integrases": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mvirs.integrases.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "summary": [
                    [
                        {
                            "id": "test"
                        },
                        "test.mvirs.summary.tsv:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a98833d0eaa28fe37eb1685b05b4c8a8"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-10-25T11:20:39.520294388"
    }
}